sklego/neighbors.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.neighbors import KernelDensity
+from sklearn.utils import check_X_y
+from sklearn.utils.multiclass import unique_labels
+from sklearn.utils.validation import check_is_fitted, check_array, FLOAT_DTYPES
+
+
+class BayesianKernelDensityClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(
+        self,
+        bandwidth=0.2,
+        kernel="gaussian",
+        algorithm="auto",
+        metric="euclidean",
+        atol=0,
+        rtol=0,
+        breath_first=True,
+        leaf_size=40,
+        metric_params=None,
+    ):
+        """
+        Bayesian Classifier that uses Kernel Density Estimations to generate the joint distribution
+
+        All parameters of the model are an exact copy of the parameters in scikit-learn.
+        """
+        # TODO (2/6/2020) add awesome docstring
+        self.bandwidth = bandwidth
+        self.kernel = kernel
+        self.algorithm = algorithm
+        self.metric = metric
+        self.atol = atol
+        self.rtol = rtol
+        self.breath_first = breath_first
+        self.leaf_size = leaf_size
+        self.metric_params = metric_params
+
+    def fit(self, X: np.ndarray, y: np.ndarray):
+        X, y = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+
+        self.classes_ = unique_labels(y)
+        self.models_, self.priors_logp_ = {}, {}
+        for target_label in self.classes_:
```

Comment:
Maybe not necessarily in this PR, but this seems easily parallelizable with something like joblib.

Reply:
Very good point, I think that's a good idea. I just confirmed that sklearn's kernel density estimator doesn't use any parallel components which is good. In most cases, it will merely half the runtime but if you have a lot of classes it could be a great benefit.